src/solver/algorithm/strong/cyclic.rs

```diff
@@ -6,3 +6,379 @@
+        if child_count > 0 {
+            let mut buf = RecordBuffer::new(game.players())
+                .context("Failed to create record for end state.")?;
+            buf.set_utility([SimpleUtility::DRAW, SimpleUtility::DRAW])?;
+            db.put(parent, &buf);
+        }
+    }
+
+    Ok(())
+}
+
+/// Set up the initial frontiers and primitive position database entries
+fn enqueue_children<G, D>(
+    winning_frontier: &mut VecDeque<State>,
+    tying_frontier: &mut VecDeque<State>,
+    losing_frontier: &mut VecDeque<State>,
+    curr_state: State,
+    game: &G,
+    child_counts: &mut HashMap<State, usize>,
+    db: &mut D,
+) -> Result<()>
+where
+    G: DTransition<State> + Bounded<State> + SimpleSum<2> + Extensive<2> + Game,
+    D: KVStore,
+{
+    if game.end(curr_state) {
+        let mut buf = RecordBuffer::new(game.players())
+            .context("Failed to create placeholder record.")?;
+        buf.set_utility(game.utility(curr_state))
+            .context("Failed to copy utility values to record.")?;
+        buf.set_remoteness(0)
+            .context("Failed to set remoteness for end state.")?;
+        db.put(curr_state, &buf);
+
+        match game
+            .utility(curr_state)
+            .get(game.turn(curr_state))
+        {
+            Some(SimpleUtility::WIN) => winning_frontier.push_back(curr_state),
+            Some(SimpleUtility::TIE) => tying_frontier.push_back(curr_state),
+            Some(SimpleUtility::LOSE) => losing_frontier.push_back(curr_state),
+            _ => {
+                panic!["Utility for primitive ending position found to be draw"]
+            },
+        }
+        return Ok(());
+    }
+
+    // Enqueue primitive positions into frontiers
+    let children = game.prograde(curr_state);
+    child_counts.insert(curr_state, children.len());
+
+    for child in children {
+        if child_counts.contains_key(&child) {
+            continue;
+        }
+        enqueue_children(
+            winning_frontier,
+            tying_frontier,
+            losing_frontier,
+            child,
+            game,
+            child_counts,
+            db,
+        )?;
+    }
+
+    Ok(())
+}
+
+/* DATABASE INITIALIZATION */
+
+/// Initializes a volatile database, creating a table schema according to the
+/// solver record layout, initializing a table with that schema, and switching
+/// to that table before returning the database handle.
+fn volatile_database<G>(game: &G) -> Result<volatile::Database>
+where
+    G: Extensive<2> + Game,
+{
+    let id = game.id();
+    let db = volatile::Database::initialize();
+
+    let schema = RecordType::SUR(2)
+        .try_into()
+        .context("Failed to create table schema for solver records.")?;
+    db.create_table(&id, schema)
+        .context("Failed to create database table for solution set.")?;
+    db.select_table(&id)
+        .context("Failed to select solution set database table.")?;
+
+    Ok(db)
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
It looks like you're using a bespoke `GameGraph` here -- please replace this with the existing `game::mock::Session` implementation (see `game::mock::builder` for how to make one).

This is probably from back when we had not finished it, so apologies for the delays on that.

Coordinate with @BnjmnZmmrmn if you would like him to make bigger test cases for you.